src/phoenix/trace/otel/decode.py

```diff
@@ -0,0 +1,203 @@
+    end_time = (
+        decode_unix_nano(otlp_span.end_time_unix_nano) if otlp_span.end_time_unix_nano else None
+    )
+
+    attributes = dict(decode_key_values(otlp_span.attributes))
+    span_kind = SpanKind(attributes.pop(PHOENIX_SPAN_KIND, None))
+
+    (
+        events,
+        documents,
+        embeddings,
+        input_messages,
+        output_messages,
+        prompt_template_variables,
+    ) = _decode_events(otlp_span.events)
+    if documents:
+        attributes[RETRIEVAL_DOCUMENTS] = documents
+    if embeddings:
+        attributes[EMBEDDING_EMBEDDINGS] = embeddings
+    if input_messages:
+        attributes[LLM_INPUT_MESSAGES] = input_messages
+    if output_messages:
+        attributes[LLM_OUTPUT_MESSAGES] = output_messages
+    if prompt_template_variables:
+        attributes[LLM_PROMPT_TEMPLATE_VARIABLES] = prompt_template_variables
+
+    status_code, status_message = _decode_status(otlp_span.status)
+
+    return Span(
+        name=otlp_span.name,
+        context=SpanContext(
+            trace_id=trace_id,
+            span_id=span_id,
+        ),
+        parent_id=parent_id,
+        start_time=start_time,
+        end_time=end_time,
+        attributes=attributes,
+        span_kind=span_kind,
+        status_code=status_code,
+        status_message=status_message,
+        events=events,
+        conversation=None,
+    )
+
+
+def _decode_identifier(identifier: bytes) -> Optional[UUID]:
+    if not identifier:
+        return None
+    try:
+        return UUID(bytes=identifier)
+    except ValueError:
+        pass
+    # Based on the int-to-bytes implementation in OpenTelemetry Python SDK. See:
+    # https://github.com/open-telemetry/opentelemetry-python/blob/3dfe2249cc4a203bf24578483b192fec7266596b/exporter/opentelemetry-exporter-otlp-proto-common/src/opentelemetry/exporter/otlp/proto/common/_internal/__init__.py#L88-L93  # noqa: E501
+    # Ultimately, we should probably stop using UUID altogether, and just use bytes everywhere.
+    return UUID(int=int.from_bytes(identifier, byteorder="big"))
+
+
+Document: TypeAlias = Dict[str, Any]
+Embedding: TypeAlias = Dict[str, Any]
+PromptTemplateVariables: TypeAlias = Dict[str, Any]
+OutputMessage: TypeAlias = Dict[str, Any]
+InputMessage: TypeAlias = Dict[str, Any]
+
+
+class SortableObject(NamedTuple):
+    time_unix_nano: int
+    tie_breaker: int
+    key_values: Iterable[Tuple[str, Any]]
+
+
+@dataclass
+class SortableObjects:
+    json_load_keys: Container[str] = field(default=())
+    objects: List[SortableObject] = field(init=False, default_factory=list)
+
+    def append(self, pb_event: otlp.Span.Event) -> None:
+        self.objects.append(
+            SortableObject(
+                time_unix_nano=pb_event.time_unix_nano,
+                tie_breaker=len(self.objects),
+                key_values=decode_key_values(pb_event.attributes, self.json_load_keys),
+            )
+        )
```

Comment:
Suggestion: I think it's more straightforward to use a list of tuples or NamedTuples rather than introducing a new data structure here.